File: app/src/main/java/com/alfianyusufabdullah/chatyuk/MyApplication.java
package com.alfianyusufabdullah.chatyuk;

import android.app.Application;
import android.content.Context;
import android.content.SharedPreferences;
import android.view.inputmethod.InputMethodManager;
import android.widget.EditText;

import com.google.firebase.database.DatabaseReference;
import com.google.firebase.database.FirebaseDatabase;

/**
 * Created by JonesRandom on 9/15/17.
 *
 * @site www.androidexample.web.id
 * @github @alfianyusufabdullah
 *
 */

public class MyApplication extends Application {

    private static MyApplication application;
    private static FirebaseDatabase database;

    @Override
    public void onCreate() {
        super.onCreate();
        application = this;
        database = FirebaseDatabase.getInstance();
        database.setPersistenceEnabled(true);

    }

    public static MyApplication getInstance(){
        return application;
    }

    public static MyApplication hideSoftInput(Context context , EditText editText){
        InputMethodManager im = (InputMethodManager)context.getSystemService(INPUT_METHOD_SERVICE);
        if (im != null) {
            im.hideSoftInputFromWindow(editText.getWindowToken() , InputMethodManager.RESULT_UNCHANGED_SHOWN);
        }
        return application;
    }

    public SharedPreferences getSharedPreferences(){
        return getSharedPreferences("user" , MODE_PRIVATE);
    }

    public static DatabaseReference getFirebaseDatabaseReferences(String Reference){
        return database.getReference(Reference);
    }

}

File: app/src/main/java/com/alfianyusufabdullah/chatyuk/MyApplication.kt
package com.alfianyusufabdullah.chatyuk

import android.app.Application
import android.content.Context
import android.view.inputmethod.InputMethodManager
import android.widget.EditText
import com.google.firebase.database.DatabaseReference
import com.google.firebase.database.FirebaseDatabase

/**
 * Created by JonesRandom on 9/15/17.
 *
 * @site www.androidexample.web.id
 * @github @alfianyusufabdullah
 */

class MyApplication : Application() {

    override fun onCreate() {
        super.onCreate()
        database = FirebaseDatabase.getInstance()
        database.setPersistenceEnabled(true)
    }

    companion object {
        private lateinit var database: FirebaseDatabase

        fun hideSoftInput(context: Context, editText: EditText) {
            val im = context.getSystemService(Context.INPUT_METHOD_SERVICE) as InputMethodManager
            im.hideSoftInputFromWindow(editText.windowToken, InputMethodManager.RESULT_UNCHANGED_SHOWN)
        }

        fun getFirebaseDatabaseReferences(Reference: String): DatabaseReference {
            return database.getReference(Reference)
        }
    }

}
